src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector512.cs

```diff
@@ -2950,6 +2950,84 @@ public static Vector512<T> WithUpper<T>(this Vector512<T> vector, Vector256<T> v
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector512<T> Xor<T>(Vector512<T> left, Vector512<T> right) => left ^ right;
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int GetMatchCount<T>(this Vector512<T> vector)
+        {
+            if (Vector512.IsHardwareAccelerated)
+            {
+                return BitOperations.PopCount(vector.ExtractMostSignificantBits());
+            }
+            else if (Vector256.IsHardwareAccelerated)
+            {
+                Vector256<T> lower = vector._lower;
+                Vector256<T> upper = vector._upper;
+
+                int lowerCount = Vector256.GetMatchCount(lower);
+                int upperCount = Vector256.GetMatchCount(upper);
+
+                return lowerCount + upperCount;
+            }
+
+            Vector128<T> vec0 = vector._lower._lower;
+            Vector128<T> vec1 = vector._lower._upper;
+            Vector128<T> vec2 = vector._upper._lower;
+            Vector128<T> vec3 = vector._upper._upper;
+
+            int cnt0 = Vector128.GetMatchCount(vec0);
+            int cnt1 = Vector128.GetMatchCount(vec1);
+            int cnt2 = Vector128.GetMatchCount(vec2);
+            int cnt3 = Vector128.GetMatchCount(vec3);
+
+            return cnt0 + cnt1 + cnt2 + cnt3;
```

Comment:
The `Vector512` fallbacks should probably not have separate V128 versions, just V256 ones.

Reply:
Let me check if the 128b fallback codegen is good enough.